src/Surfnet/Stepup/Identity/Value/UnhashedSecret.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * Copyright 2022 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Stepup\Identity\Value;
+
+/**
+ * Unhashed secret
+ *
+ * This VO can be used to hash the unhashed secret
+ *
+ * A HashedSecret is constructed when calling the immutable hashSecret
+ * method. This method will return an instance of HashedSecret.
+ */
+class UnhashedSecret implements HashableSecret
+{
+    /**
+     * Cost to compute the hash (10 is the default baseline value for hash_password)
+     * Increasing this value also increases the time to generate it.
+     */
+    private const COST = 14;
```

Comment:
I suspect you have taken into account that no rehashing is implemented and that the cost will be sufficient over time?